### PR TITLE
Quick buttons work fix and insert step functionality corrected

### PR DESCRIPTION
**Pull request by Divy123**
Fixes #0000  (<=== Replace `0000` with the Issue Number)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `npm test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts
* [ ] PR is descriptively titled
* [ ] ask `@publiclab/is-reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

Thanks!

### CODE REVIEW

**Comment by Divy123** [comment]
@jywarren this is fixed.

Now, I will proceed towards UI testing in one of my other PRs and would request for help from @aashna27 for the same.

**Comment by HarshKhandeparkar** [comment]
@Divy123 can this be merged after my scopeQuery PR gets merged as I won't have time to rebase it later on?

**Comment by Divy123** [comment]
Ya sure!!

I have already synced my PR with yours so I don't have any issues with that.

File: examples/lib/defaultHtmlSequencerUi.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ function DefaultHtmlSequencerUi(_sequencer, options) {
 
   function onLoad() {
     importStepsFromUrlHash();
-    if ($('#selectStep').val() === 'none')
+    if (!$('#selectStep').val())
```
Comment (IgorWilbert) [inline examples/lib/defaultHtmlSequencerUi.js]: 👌

File: examples/lib/intermediateHtmlStepUi.js [hunk 2]
```diff
@@ -128,46 +116,42 @@ function IntermediateHtmlStepUi(_sequencer, step, options) {
...
-    insertStepSelect.selectize({
-      sortField: 'text'
-    });
-    //$step('.inserDiv .add-step-btn').prop('disabled', true);
-    $(step.ui.querySelector('.inserDiv .add-step-btn')).prop('disabled', true);
+
+    $('.inserDiv .add-step-btn').prop('disabled', true);
     
     insertStepSelect.append('<option value="" disabled selected>Select a Module</option>');
     //$step('.insertDiv .radio-group .radio').on('click', function () {
```
Comment (IgorWilbert) [inline examples/lib/intermediateHtmlStepUi.js]: Can we get rid of this one too? `//$step('.insertDiv .radio-group .radio').on('click', function () {`

**Comment by HarshKhandeparkar** [comment]
Thank you!

File: examples/demo.js [hunk 3]
```diff
@@ -24,9 +24,6 @@ window.onload = function() {
     }
     // Null option
     addStepSelect.append('<option value="" disabled selected>Select a Module</option>');
-    addStepSelect.selectize({
```
Comment (IgorWilbert) [inline examples/demo.js]: @Divy123 it is not clear to me why we no longer need this block, could you please explain?

File: examples/lib/intermediateHtmlStepUi.js (hunk 2 above)
Comment (Divy123) [reply examples/lib/intermediateHtmlStepUi.js]: Ya thanks will do that.

File: examples/demo.js (hunk 3 above)
Comment (Divy123) [reply examples/demo.js]: Actually this implemented search functionality but this is a big error causing agent and is not so good. So as of now I have fixed all the code and will implement a search functionality as soon as we find something apt.

**Comment by Divy123** [comment]
@jywarren please review this.

**Comment by Divy123** [comment]
@jywarren please review this.

**Comment by jywarren** [comment]
Oof, can this be rebased cleanly first? Sorry to ask, but there are so many merges!

**Comment by jywarren** [comment]
It looks good though... but hard to assess! This'll be much easier to review once we have UI tests that we can point at that put this all to use. Thanks!

**Comment by Divy123** [comment]
Ya the merges were made to keep in sync with Harsh's PR . And as of now everything is in sync. Also, I am not so good at rebasing, that I am afraid to break something. Can you please help me with that?

**Comment by Divy123** [comment]
On the part of UI tests, I am writing a few of them in one of my own repos and then come here but if you say, I can share a working GIF here to confirm the working and since this is a major bug fix,I  think should be merged soon.

**Comment by jywarren** [comment]
For rebasing, have you read through here? https://publiclab.org/developers#Resources

But, if you're worried, try opening a new PR for the rebase instead of overwriting this branch. Then you can compare the "files changed" to confirm that the resulting changes are the same!

**Comment by jywarren** [comment]
OK, yes, let's just merge after a rebase, thank you!!!

**Comment by Divy123** [comment]
Thanks a  lot!!

**Comment by Divy123** [comment]
@jywarren I have done the merges already, can you please guide me on how can I rebase or should I open a separate PR?

**Comment by jywarren** [comment]
Hi, i still see this:

image

Would you like to try rebasing, following the guidance under "Git" here? https://publiclab.org/developers#Resources

**Comment by Divy123** [comment]
Can I go with a merge fix?

**Comment by jywarren** [comment]
sure as it'll be rebased when we squash and merge. I just couldn't tell how
you wanted to resolve the conflicts.

**Comment by Divy123** [comment]
@jywarren this is updated!!

**Comment by Divy123** [comment]
@jywarren can we merge it now?

**Comment by Divy123** [comment]
@jywarren please review!!

**Comment by Divy123** [comment]
@jywarren this is ready to be merged here

**Comment by jywarren** [comment]
Hmm, GitHub still says it's out of date... But I'll try to update. Then we can merge. I'm sorry I missed this! Thanks for your patience!

File: examples/lib/intermediateHtmlStepUi.js [hunk 4]
```diff
@@ -2,9 +2,9 @@ var urlHash = require('./urlHash.js'),
   insertPreview = require('./insertPreview.js');
 
 function IntermediateHtmlStepUi(_sequencer, step, options) {
-  function stepUI() {
-    return '<div class="row insertDiv collapse">\
-          <section class="panel panel-primary .insert-step">\
+  function stepUI(id) {
+    return '<div class="row insertDiv collapse index' + id + '">\
```
Comment (jywarren) [inline examples/lib/intermediateHtmlStepUi.js]: Here maybe index+id could be named a bit more specifically?

Comment (Divy123) [reply examples/lib/intermediateHtmlStepUi.js]: So basically, I am using id as a variable and index just to signify its meaning for the code readability.

**Comment by Divy123** [comment]
Ya sure, I am in discussing with @aashna27 on how to achieve UI testing the best in this case.

**Comment by jywarren** [comment]
Hi, @Divy123 any updates here? Thank you!

**Comment by HarshKhandeparkar** [comment]
Fixed elsewhere.

**Pull request closed without merging**